### PR TITLE
Seba individual card

### PR DESCRIPTION
**Pull request by sebkaluzny**

### CODE REVIEW

File: app/src/main/java/com/example/android/psiak/model/DogFirebase.java [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import java.util.List;
 
 @IgnoreExtraProperties
-public class DogFirebase {
+public class DogFirebase implements AnimalInterface {
```
Comment (mbialorucki) [inline app/src/main/java/com/example/android/psiak/model/DogFirebase.java]: This should be class (maybe abstract?)

File: app/src/main/java/com/example/android/psiak/data/network/FirebaseRepository.java [hunk 2]
```diff
@@ -18,135 +18,157 @@
...
-    @Override
-    public void update(DogFirebase firebaseObject) {
-        Map<String, Object> childUpdates = new HashMap<>();
-        childUpdates.put(firebaseObject.getId(), firebaseObject);
-        dogsReference.updateChildren(childUpdates, new DatabaseReference.CompletionListener() {
-            @Override
-            public void onComplete(DatabaseError databaseError, DatabaseReference databaseReference) {
-                // TODO: Inform user that data was updated successfully
+  @Override
+  public void getById(String id, final FirebaseDataListener callback) {
```
Comment (mbialorucki) [inline app/src/main/java/com/example/android/psiak/data/network/FirebaseRepository.java]: Formatting : one dot per line

File: app/src/main/java/com/example/android/psiak/ui/main/MainPresenter.java [hunk 3]
```diff
@@ -13,51 +14,57 @@
  */
 
 class MainPresenter
-        extends BasePresenter<MainContract.View>
-        implements MainContract.Presenter<MainContract.View> , FirebaseDataListener {
+    extends BasePresenter<MainContract.View>
+    implements MainContract.Presenter<MainContract.View>, FirebaseDataListener {
```
Comment (mbialorucki) [inline app/src/main/java/com/example/android/psiak/ui/main/MainPresenter.java]: Presenter shouldn't be coupled with Firebase implementation. Maybe new class that will encapsulate Firebase implementation details?

File: app/src/main/java/com/example/android/psiak/ui/showAnimalDetails/ShowAnimalDetailsPresenter.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+  public ShowAnimalDetailsPresenter(Repository.Firebase<DogFirebase> repository) {
+    this.repository = repository;
+  }
+
+  @Override
+  public void loadAnimal(String id) {
+    view.showLoader();
+
+    this.repository.getById(id, new FirebaseDataListener() {
```
Comment (mbialorucki) [inline app/src/main/java/com/example/android/psiak/ui/showAnimalDetails/ShowAnimalDetailsPresenter.java]: Presenter shouldn't know anything about data layer implementation details.

**Review by PawelZarzycki — DISMISSED** [review]
ok

**Review by GrzegorzKwasniewski — DISMISSED** [review]

File: app/src/main/java/com/example/android/psiak/data/local/InMemoryDogRepository.java [hunk 5]
```diff
@@ -14,17 +13,16 @@
...
     }
 
-    private InMemoryDogRepository(){
-        this.items = new ArrayList<>();
-    }
-
-
     @Override
     public List<Dog> getAll() {
         return items;
```
Comment (sebkaluzny) [inline app/src/main/java/com/example/android/psiak/data/local/InMemoryDogRepository.java:28]: irrelevant editor changes

File: app/src/main/java/com/example/android/psiak/data/network/DummyDogDataService.java [hunk 6]
```diff
@@ -10,6 +9,6 @@
 public interface DummyDogDataService {
 
     @GET("/bins/10jqvj")
-    Call<List<Dog>> loadDog();
+    Call<List<DogFirebase>> loadDog();
 
 }
```
Comment (sebkaluzny) [inline app/src/main/java/com/example/android/psiak/data/network/DummyDogDataService.java:14]: irrelevant changes

**Review by PawelZarzycki — DISMISSED** [review]
ok

**Review by GrzegorzKwasniewski — APPROVED** [review]

**Pull request merged**